feat(extxyz): add unit convert and tag synonym matching

SchrodingersCattt opened the pull request:
Many datasets using the `exyz` format have different formats for units and tags. This PR aims to address the situation with multiple possible datasets.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added unit-aware parsing for extended XYZ header metadata, automatically converting energies, forces, and stress/virials based on declared `energy-unit`, `force-unit`, and `stress-unit`.

- **Improvements**
  - Consistent conversion when deriving virials from stress and when applying `virials` with an `energy-unit`; supports common unit shorthands including atomic units (`au`/`a.u.`), and composites (e.g., `ev/angstrom^3`). Invalid/unsupported units now fail fast.

- **Tests**
  - Expanded unit-conversion and integration-style coverage with new fixtures (hartree, kcal/mol, GPa, kbar, atomic units, unsupported units, virials in hartree).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

SchrodingersCattt committed:
quipGapXYZ: add unit convert and synonym matching

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

SchrodingersCattt committed:
style: add Exception after except

SchrodingersCattt committed:
style: add  after

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

SchrodingersCattt committed:
fix: removed redundant

SchrodingersCattt committed:
style: add Exception after except

njzjz commented on dpdata/xyz/quip_gap_xyz.py:
```diff
@@ -82,56 +95,72 @@ def handle_single_xyz_frame(lines):
+                    used_colomn += field_length
+                    continue
+            except Exception as e:
+                print("unknown field {}".format(kv_dict["key"]), e)
```
What is the reason not to raise an error?

njzjz commented on dpdata/xyz/quip_gap_xyz.py:
```diff
@@ -82,56 +95,72 @@ def handle_single_xyz_frame(lines):
+                    ]
+                    used_colomn += field_length
+                    continue
+            except Exception as e:
```
What is the error it is expected to catch?

njzjz commented on dpdata/xyz/quip_gap_xyz.py:
```diff
@@ -165,21 +194,111 @@ def handle_single_xyz_frame(lines):
-        ).astype("float32")
+
         info_dict["coords"] = np.array([coords_array]).astype("float32")
+        """
         info_dict["energies"] = np.array([field_dict["energy"]]).astype("float32")
         info_dict["forces"] = np.array([force_array]).astype("float32")
+        """
```
If it is not used anymore, it should be removed.

njzjz commented on dpdata/xyz/quip_gap_xyz.py:
```diff
@@ -165,21 +194,111 @@ def handle_single_xyz_frame(lines):
+                info_dict["energies"] = np.array([field_dict["energy"]]).astype(
+                    "float32"
+                )
+        except Exception:
```
What is the expected error to catch here? It's hard to read.

njzjz commented on dpdata/xyz/quip_gap_xyz.py:
```diff
@@ -165,21 +194,111 @@ def handle_single_xyz_frame(lines):
+                )
+            elif e_units == "ev":
+                info_dict["energies"] = np.array([field_dict["energy"]]).astype(
+                    "float32"
```
Why is it converted to float32?

njzjz reviewed (changes requested):
The unit test should be added to cover the cases in this PR.

SchrodingersCattt committed:
feat(extxyz): add table-driven unit conversion for energy, force, and stress

- New module dpdata/formats/xyz/_unit_convert.py with unit alias maps
  and _get_unit_factor() helper that delegates to dpdata.unit converters
- Support energy-unit, force-unit, stress-unit headers in extxyz files
- Convert to dpdata internal units (eV, eV/angstrom, eV/angstrom^3)
- Raise ValueError for unsupported units instead of silent pass
- Use float64 throughout instead of float32
- Add _parse_force_unit() for composite unit strings (e.g. kcal/mol/angstrom)
- Add stress_factor parameter to _parse_stress_to_virials()
- Add comprehensive tests for unit conversion module and end-to-end

Resolves reviewer feedback from PR #678:
- No more bare except / silent error swallowing
- No more dead commented-out code
- No more float32 precision truncation
- Full test coverage for unit conversion paths

SchrodingersCattt committed:
Merge remote-tracking branch 'upstream/master' into pr-678

# Conflicts:
#	dpdata/xyz/quip_gap_xyz.py

SchrodingersCattt committed:
fix(extxyz): convert virial with energy-unit factor and add au/a.u. aliases

- Apply e_factor to explicit virial= values (virial has energy dimensions)
- Add 'au' and 'a.u.' as aliases for hartree in _ENERGY_UNIT_MAP
- Handle force-unit=au/a.u. as hartree/bohr in _parse_force_unit
- Add regression tests for non-eV virial and atomic-unit aliases

SchrodingersCattt committed:
feat(extxyz): log unit conversions when factor != 1.0

When energy-unit, force-unit, or stress-unit headers trigger a non-trivial
conversion, emit an INFO-level log message showing the source unit and
conversion factor for traceability.

SchrodingersCattt committed:
fix(extxyz): add au/a.u. stress-unit aliases and Pa/kPa support

- Add 'au/bohr^3' and 'a.u./bohr^3' to _PRESSURE_UNIT_MAP (→ hartree/bohr^3)
- Add 'pa' and 'kpa' to _PRESSURE_UNIT_MAP for completeness
- Add regression test for stress-unit=au/bohr^3 → virial conversion
